### DISCUSSION TITLE
need help with math in rules

### DISCUSSION THREAD

**Discussion by pacoso81** [arendst/Tasmota#15665] [Q&A] [answered]
a have setup an simple thermostat using rules.
rule1
ON TempTargetSet#data do backlog Mem1 %value%; var1 %value%; var2 %value% ENDON 

rule2
ON DS18B20#temperature>%var1% DO backlog var2 %var1%; var1 99; Power 0 ENDON
ON DS18B20#temperature<%var2% DO backlog var1 %var2%; var2 -99; Power 1 ENDON

i want to add hysteresis in the var1 or var2 so that when DS temperature is near the setpoint i don't get Power 0, Power1, .....
for example if i set the setpoint to 40 i want var1 to be 40 and var2 to be 38. 
i don't have much experience whit math in rules, i don't know if it is even possible.

**Comment by sfromis** [comment, marked as answer]
Here's an example of sensor hysteresis in rules.
https://tasmota.github.io/docs/Rules/#transmit-sensor-value-only-when-a-delta-is-reached
